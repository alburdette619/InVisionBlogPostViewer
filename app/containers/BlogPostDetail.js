import React from 'react';
import PropTypes from 'prop-types';
import { WebView } from 'react-native';

const BlogPostDetail = (props) => {
  const { postUrl } = props.navigation.state.params;
  return (<WebView source={{ uri: postUrl }} />);
};


BlogPostDetail.propTypes = {
  navigation: PropTypes.object.isRequired,
};

export default BlogPostDetail;
