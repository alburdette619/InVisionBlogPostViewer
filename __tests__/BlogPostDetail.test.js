import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import BlogPostDetail from '../app/containers/BlogPostDetail';

describe('Test BlogPostDetail', () => {
  it('should render BlogPostDetail correctly', () => {
    const detail = renderer
      .create(<BlogPostDetail
        navigation={{
            state: {
              params: {
                postUrl:
                  'http://engineering.invisionapp.com/post/painless-microservices-migration/',
              },
            },
          }}
      />)
      .toJSON();
    expect(detail).toMatchSnapshot();
  });
});
