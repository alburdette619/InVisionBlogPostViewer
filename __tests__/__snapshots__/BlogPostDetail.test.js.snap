// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test BlogPostDetail should render BlogPostDetail correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTWebView
    allowsInlineMediaPlayback={undefined}
    automaticallyAdjustContentInsets={undefined}
    bounces={undefined}
    contentInset={undefined}
    dataDetectorTypes={undefined}
    decelerationRate={undefined}
    injectedJavaScript={undefined}
    mediaPlaybackRequiresUserAction={undefined}
    messagingEnabled={false}
    onLoadingError={[Function]}
    onLoadingFinish={[Function]}
    onLoadingStart={[Function]}
    onMessage={[Function]}
    onShouldStartLoadWithRequest={undefined}
    scalesPageToFit={true}
    scrollEnabled={undefined}
    source={
      Object {
        "uri": "http://engineering.invisionapp.com/post/painless-microservices-migration/",
      }
    }
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "backgroundColor": "#ffffff",
        },
        undefined,
      ]
    }
  />
</View>
`;
